container/src/utilities/helpers.ts

```diff
@@ -21,22 +21,32 @@ export class GenericHelpersClass {
 
   /**
    * Checks whether web component is an attribute or property. In case of attribute, it returns the parsed value.
-   * @param webcomponent 
-   * @returns returns the parsed webcomponent value.
+   * @param {object | boolean | string} webcomponent value can either be an object, boolean or a stringified object, e.g webcomponent='{"selfregistered":"true"}'
+   * @returns {object | boolean} webcomponent returns the parsed webcomponent value.
    */
-  checkWebcomponentValue(webcomponent: any): object | boolean {
+  checkWebcomponentValue(webcomponent: object | boolean | string): object | boolean {
     if (typeof webcomponent === 'string') {
-      let parsedValue = JSON.parse(webcomponent);
-      if (this.isObject(parsedValue)) {
-        if ('selfRegistered' in parsedValue) {
-          parsedValue.selfRegistered = parsedValue.selfRegistered === 'true';
-        }
-      }
-      return parsedValue;
+      return JSON.parse(webcomponent);
     } else if (typeof webcomponent === 'boolean' || typeof webcomponent === 'object') {
       return webcomponent;
+    } else {
+      console.warn('Webcomponent value has a wrong type.')
     }
   }
+
+  /**
+  * Resolves the context to an object. If the context is a string, it attempts to parse
+  * it as JSON. If parsing fails JSON parse error will be thrown.
+  * @param {object | string} context - The context to be resolved.
+  * @returns {object} The resolved context as an object.
+  */
+  resolveContext(context: object | string): object {
+    return context
+      ? typeof context === 'string'
+        ? JSON.parse(context)
```

Comment:
It certainly doesn't hurt to have **Json.parse** fail as we can catch this in our unit tests easily as a native JSON thrown error. 

However I think this pollutes the Console with seemingly **unhandled error messages**, making it look like the error is on our side. 
It would look better if we actually handle the thrown error with a **try/catch** statement and a default warning and default **{} return value** on failure instead of letting **JSON.parse** throw whatever error wants.

Reply:
The reason why I want to avoid the try/catch is that the luigiContainer or luigiCompoundContainer should fail if there is an error. When we do try/catch the wc will shown although there is a wrong context value. This can lead to unforeseen behaviour, wc is there but with a wrong context. In addition besides the property/attribute _webcomponent_ all the others will not validated as well.
For example if you have somthing like this:
```
<luigi-container id="lc" data-test-id="luigi-container" client-permissions="test"></luigi-container>
```
you will get an invalid json as well.

Reply:
I see 🤔 . In that case we could also do a custom throw Error to have the same intended fail behaviour but make it easier for devs to find where they did wrong. Something like:
```
try {

} catch(Error e) {
 console.error("Failed because of X specific reason");
 throw Error(e);
}
```
But I guess we'd have to discuss it first if its too ugly or too much detailed

Reply:
In that case we need to implement it for every property/attribute, because we can't do it for context and webcomponent only.

Let's discuss it in the daily.